### PR TITLE
Updating viewer to use latest vcr and imodels-* pks supporting itwinjs core 4.0

### PR DESCRIPTION
**Pull request by Jake-Screen**
- Updated viewer-components-react packages in web/desktop viewer and their templates to use latest versions that support itwinjs core 4.0.0 official release
-  Updated imodels-client packages in web/desktop viewer and their templates to use latest versions that support itwinjs core 4.0.0 official release
-  Updated `@itwin/electron-authorization` to 0.14.0 in desktop-viewer-test and desktop-viewer-react
-  Updated `@itwin/browser-authorization` to 0.14.0 in web-viewer-test and web-viewer-react
- fixed runtime error in desktop viewer by changing argument types and arguments in ipcCall function 
- Updated files in desktop template to bring it in line with desktop-viewer-test, mainly linting changes and electron authorization updates
- Updated to electron 24 in desktop viewer packages
- Reverted README files for templates to use npm commands

### CODE REVIEW

**Review by aruniverse — COMMENTED** [review]
does this fix the runtime issue with ipc calls?

**Comment by Jake-Screen** [comment]
> does this fix the runtime issue with ipc calls?

No, still seeing the same issue.

**Comment by Jake-Screen** [comment]
ipc runtime issue fixed

**Comment by aruniverse** [comment]
@linasburneika mentioned:

```
Hi, I noticed that @itwin/viewer-react@4.0.0-dev.14 uses @itwin/presentation-components but this dependency is not listed in package.json

file UnifiedSelectionViewportControl:
import { viewWithUnifiedSelection } from "@itwin/presentation-components";
```

can you also update to electron 24 for desktop too?

**Comment by Jake-Screen** [comment]
@aruniverse Updated to electron 24. viewer-react does have `@itwin/presentation-components` listed in its package.json on the next branch and in this pr

File: packages/templates/cra-template-desktop-viewer/template/src/frontend/components/modelSelector/SelectIModel.tsx [hunk 1]
```diff
@@ -50,7 +50,7 @@ const useProgressIndicator = (iModel: IModelFull) => {
         );
       });
     }
-  }, [userSettings]);
+  }, [userSettings, iModel.id, iModel.projectId]);
```
Comment (aruniverse) [inline packages/templates/cra-template-desktop-viewer/template/src/frontend/components/modelSelector/SelectIModel.tsx:53]: were the changes in these files also made in the desktop test app here?

Comment (Jake-Screen) [reply packages/templates/cra-template-desktop-viewer/template/src/frontend/components/modelSelector/SelectIModel.tsx:53]: yes, the changes came from there originally

File: packages/templates/cra-template-desktop-viewer/template.json [hunk 2]
```diff
@@ -73,6 +74,8 @@
         "semver": "7.4.0"
       }
     },
-    "browserslist": ["electron 23.0.0"]
+    "browserslist": [
+      "electron 23.0.0"
```
[the remark this answers is not in the document]
Comment (Jake-Screen) [reply packages/templates/cra-template-desktop-viewer/template.json]: changed

File: packages/templates/cra-template-web-viewer/template.json [hunk 3]
```diff
@@ -48,10 +48,8 @@
       "test": "react-scripts --max_old_space_size=4096 test",
       "eject": "react-scripts eject"
     },
-    "pnpm": {
-      "overrides": {
-        "semver": "7.4.0"
-      }
+    "overrides": {
```
Comment (aruniverse) [inline packages/templates/cra-template-web-viewer/template.json:51]: delete this chunk

Comment (Jake-Screen) [reply packages/templates/cra-template-web-viewer/template.json]: removed

**Review by aruniverse — APPROVED** [review]

**Pull request merged**